app/src/main/java/io/apicurio/registry/rules/compatibility/CompatibilityRuleExecutor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.rules.compatibility;
+
+import java.util.Collections;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.apicurio.registry.rules.RuleContext;
+import io.apicurio.registry.rules.RuleExecutor;
+import io.apicurio.registry.rules.RuleViolationException;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+public class CompatibilityRuleExecutor implements RuleExecutor {
+
+    /**
+     * @see io.apicurio.registry.rules.RuleExecutor#execute(io.apicurio.registry.rules.RuleContext)
+     */
+    @Override
+    public void execute(RuleContext context) throws RuleViolationException {
+        CompatibilityLevel level = CompatibilityLevel.valueOf(context.getConfiguration());
```

Comment:
The `RuleConfigurationDTO` is a single String value stored in the database, converted here into compatibility level. It can have a different use for configuring e.g. a validation rule. The RuleContext has only one such configuration field but it looks like it'll contain configuration for all the rules that are going to be applied for a specified artifact update.
I think the flow of data here is not very clear and visible... maybe the RuleContext should already contain the `CompatibilityLevel` as a separate field.

Reply:
I added the context so we don't have N parameters, but have all this possible / different parameters in this context class.
Some might be null or have default values, for different purposes.

Reply:
We create a new `RuleContext` instance for each rule that is executed/applied.  So I think it's ok.